projects/opentrain/gtfs_pipeline/heatmap/HeatMap.java

```diff
@@ -0,0 +1,895 @@
+    heatMap =
+        new double[(int) (maxLatitude * 100) * 2]
+            [(int) (maxLongitude * 100) * 2]; // *2 to up and down, *2 right and left
+
+    for (StopsAround stp : stopsAroundList) {
+
+      // Arrival time of each stop from the chosen stop by location
+      // *100 to make difference between location
+      stp.i = (int) (stp.stop.getStopLat() * 100);
+      stp.j = (int) (stp.stop.getStopLon() * 100);
+      heatMap[stp.i][stp.j] = (int) (stp.arrivalTime);
+    }
+
+    // Distance of other points to the nearest stop
+
+    for (int i = 0; i < heatMap.length - 1; i++) {
+      for (int j = 0; j < heatMap[0].length - 1; j++) {
+
+        if (heatMap[i][j] == 0) // not a stop
+        {
+          StopsAround firstStop = stopsAroundList.get(0);
+          double ac = Math.abs(0 - firstStop.i); // y distance
+          double cb = Math.abs(0 - firstStop.j); // x distance
+          double distance = Math.hypot(ac, cb);
+          double stopTimeArrival = firstStop.arrivalTime;
+
+          // Find nearest stop
+          for (StopsAround stp : stopsAroundList) {
+
+            ac = Math.abs(i - stp.i); // y distance
+            cb = Math.abs(j - stp.j); // x distance
+
+            if (distance > Math.hypot(ac, cb)) {
+              distance = Math.hypot(ac, cb);
+              stopTimeArrival = stp.arrivalTime;
+            }
+          }
+          // *60 minutes to seconds
+          heatMap[i][j] = (int) ((distance * 60) + stopTimeArrival);
+        }
+        if (heatMap[i][j] == 0) {
+          System.out.println("Check heatMap assignment");
+        }
+      }
+    }
+  }
+
+  // Draw only the heat map colors
+  public void drawMap(String imagePath) {
+
+    // color1 to color2
+    Color color1 = new Color(255, 208, 235);
+    Color color2 = new Color(126, 0, 67);
+
+    // Normalize data
+    double maxData = 0;
+    double minData = heatMap[10][10];
+    for (int i = 0; i < heatMap.length - 1; i++) {
+      for (int j = 0; j < heatMap[0].length - 1; j++) {
+        if (maxData < heatMap[i][j]) {
+          maxData = heatMap[i][j];
+        }
+        if (heatMap[i][j] != 0 && minData > heatMap[i][j]) {
+          minData = heatMap[i][j];
+        }
+      }
+    }
+
+    // Create image /** TODO: Try different size of image
+    final BufferedImage image =
+        new BufferedImage(heatMap.length, heatMap[0].length, BufferedImage.TYPE_INT_ARGB);
+    final Graphics2D graphics2D = image.createGraphics();
+
+    // Draw the map
+    for (int i = 0; i < heatMap.length - 1; i++) {
+      for (int j = 0; j < heatMap[0].length - 1; j++) {
+        if (heatMap[i][j] == 0) {
+          System.out.println("ERROR:" + "\n" + "heatmap matrix value should not be 0");
+        }
+
+        float value = (float) (heatMap[i][j] - minData) / (float) (maxData - minData);
+
+        int r =
+            (int)
+                (((float) color2.getRed() - (float) color1.getRed()) * value
```

Comment:
I think it's enough to leave one float here, the first one.
float + int = float.